data/spells/scripts/support/levitate.lua

```diff
@@ -0,0 +1,32 @@
+local function levitate(creature, parameter)
+	local fromPosition = creature:getPosition()
+
+	if parameter == "up" and fromPosition.z ~= 8 or parameter == "down" and fromPosition.z ~= 7 then
```

Comment:
It's not so trivial to compare strings, it is generally bound by the shortest in O(n), can have cache and aligning issues and whatnot, if that's what you meant. I mean, you could have two `local UP = 1` and `local DOWN = -1` "constants" and use them.

```lua
local floorChange = (param == 'up' and UP) or (param == 'down' and DOWN) or 0
if floorChange == 0 then
    return RETURNVALUE_NOTPOSSIBLE
end

local tile = Tile(floorChange == UP and Position(fromPosition.x, fromPosition.y, fromPosition.z + floorChange) or toPosition)
if not tile or not tile:getGround() and not tile:hasFlag(floorChange == UP and TILESTATE_IMMOVABLEBLOCKSOLID or TILESTATE_BLOCKSOLID) then
```
And so on. Of course, if you feel it's not clear enough, it's just a suggestion 😛